api4/command.go

```diff
@@ -212,7 +212,13 @@ func executeCommand(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	commandArgs.TeamId = channel.TeamId
+	if commandArgs.TeamId == "" {
+		commandArgs.TeamId = channel.TeamId
+	} else if c.Session.GetTeamByTeamId(commandArgs.TeamId) == nil {
```

Comment:
This method only returns the team if you are a member of it (I think it's kind of poorly named.). So if you are not a member of the given team, it returns an error. Otherwise, execution continues.